From 42fbab715b33f51974a1b0d91115589d70b0ba97 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Wed, 15 Jul 2026 11:53:35 +0000
Subject: [PATCH] os_dep/linux: ioctl_cfg80211: adapt remain_on_channel to v7.2
 prototype

Linux 7.2 adds a trailing "const u8 *rx_addr" argument to the cfg80211
remain_on_channel op (commit 4dbd1829045e "wifi: cfg80211: Add MAC
address filter to remain_on_channel"). Extend the signature of
cfg80211_rtw_remain_on_channel() to match, guarded by kernel version.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 os_dep/linux/ioctl_cfg80211.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
index 179002d2..431392ec 100644
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -6873,7 +6873,11 @@ static s32 cfg80211_rtw_remain_on_channel(struct wiphy *wiphy,
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 8, 0))
 	enum nl80211_channel_type channel_type,
 #endif
-	unsigned int duration, u64 *cookie)
+	unsigned int duration, u64 *cookie
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 2, 0))
+	, const u8 *rx_addr
+#endif
+	)
 {
 	s32 err = 0;
 	u8 remain_ch = (u8) ieee80211_frequency_to_channel(channel->center_freq);
